### PR TITLE
DFC-191 | Add Tracking for Details Component

### PR DESCRIPTION
**Pull request by SophiaDWokoma**
### Description

The data team has requested that we add tracking for the details element to monitor user interactions and understand their usage and the pages where they occur.

This PR introduces the tracking logic for the details element along with the associated unit tests.

Tracking has been added for the GOV.UK [details](https://design-system.service.gov.uk/components/details/) component, targeting specific elements via classes. When a user toggles the details component, the tracker will activate and push data to the data layer for both open and close actions.

A branch has been created in the [di-fec-ga4-repo](https://github.com/govuk-one-login/di-fec-ga4-demo/branches) called DFC-191-test-selectContentTracker to test this tracker.

You can also view the [ticket](https://govukverify.atlassian.net/jira/software/c/projects/DFC/boards/436?search=select%20content%20&selectedIssue=DFC-191), to see the full acceptance criteria, beware this also contains stuff for accordion tracking which has been agreed won't be done due to their absence in the one login journey

**Example DL provided by Data Team**
Screenshot 2024-05-23 at 16 32 23

### Tickets

(DFC-191)[https://govukverify.atlassian.net/browse/DFC-191]

### Steps to Reproduce

1. Run `npm run build`
2. Copy the contents of dist/lib/analytics.js . 
3. Go to di-fec-ga4-demo, checkout DFC-191-test-selectContentTracker branch to test 
4. Paste code into the analytics.js file in node_modules/@govuk-one-login/one-login-analytics/lib/analytics.js file.
5. Run app and connect to GTM
6. Toggle both details components and check tracker is fired for open/close actions and DL values are correct.

### Checklist

[ ] - Are the commit messages for this PR in line with the GDS way?
[ ] - Have the changes in this PR been tested locally (if required)
[ ] - Have additional tests been added where required?
[ ] - Does the feature pass accessibility checks? (UI Components only)

### Additional Information

Screenshot 2024-05-23 at 16 29 34
Screenshot 2024-05-23 at 16 30 02

### CODE REVIEW

File: src/analytics/selectContentTracker/selectContentTracker.test.ts [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        link_domain: "undefined",
+        "link_path_parts.1": "undefined",
+        "link_path_parts.2": "undefined",
+        "link_path_parts.3": "undefined",
+        "link_path_parts.4": "undefined",
+        "link_path_parts.5": "undefined",
+      },
+    };
+
+    const dataLayerEventSecondDetails: SelectContentEventInterface = {
```
Comment (Joanneyoung01) [inline src/analytics/selectContentTracker/selectContentTracker.test.ts:94]: I think the only difference between `dataLayerEventFirstDetails` and `dataLayerEventSecondDetails` is the `text` attribute? So i think the first example is enough of a test, as there's not functionality to test the difference between

**Comment by Joanneyoung01** [comment]
Generally looks great Sophia! Well done :D I don't think we need both first and second details as there isn't any functional different between them, so can remove the second test? Also might be worth raising a PR to add those details components to the demo app :)

**Comment by Joanneyoung01** [comment]
Screenshot 2024-05-23 at 17 41 37
Screenshot 2024-05-23 at 17 41 55

Verified locally

File: src/analytics/selectContentTracker/selectContentTracker.test.ts (hunk 1 above)
Comment (SophiaDWokoma) [reply src/analytics/selectContentTracker/selectContentTracker.test.ts:94]: Hmm but I was just checking that if there are multiple details components , it is retrieving the right values for each ?

Comment (SophiaDWokoma) [reply src/analytics/selectContentTracker/selectContentTracker.test.ts:94]: So essentially verifying this loop 
Screenshot 2024-05-24 at 09 30 05

**Comment by SophiaDWokoma** [comment]
> Generally looks great Sophia! Well done :D I don't think we need both first and second details as there isn't any functional different between them, so can remove the second test? Also might be worth raising a PR to add those details components to the demo app :)

PR created[ here](https://github.com/govuk-one-login/di-fec-ga4-demo/pull/55)

File: src/analytics/selectContentTracker/selectContentTracker.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        text: this.getSummaryText(element),
+        section: "undefined",
+        action: this.getActionValue(element),
+        external: "undefined",
+        link_domain: "undefined",
+        "link_path_parts.1": "undefined",
+        "link_path_parts.2": "undefined",
+        "link_path_parts.3": "undefined",
+        "link_path_parts.4": "undefined",
+        "link_path_parts.5": "undefined",
```
Comment (di-aholme) [inline src/analytics/selectContentTracker/selectContentTracker.ts:58]: These undefined pieces of data will need to be populated. Have a look at some of the other trackers and follow the same pattern. Any issues with a particular field, flag it to Sam :)

**Review by di-aholme — CHANGES_REQUESTED** [review]

**Comment by di-aholme** [comment]
@SophiaDWokoma @Joanneyoung01 @nickhealGDS Converting this PR to a draft as we'll need to hold off on merging until post GA4 go-live as this is not considered in scope for the initial roll out at this point

File: src/analytics/selectContentTracker/selectContentTracker.ts (hunk 2 above)
Comment (SophiaDWokoma) [reply src/analytics/selectContentTracker/selectContentTracker.ts:58]: Ohh ok but on the ticket and in the implementation guide, it says should be left undefined ?


https://govukverify.atlassian.net/wiki/spaces/PI/pages/3603791886/WIP+GA4+Implementation+Guide#When-should-this-fire%3F.9

https://govukverify.atlassian.net/jira/software/c/projects/DFC/boards/436?search=select%20content%20&selectedIssue=DFC-191

Comment (nickhealGDS) [reply src/analytics/selectContentTracker/selectContentTracker.ts:63]: @di-aholme I think we confirmed with Sam that this was the correct behaviour?

**Review by nickhealGDS — APPROVED** [review]

**Pull request merged**